tools/cloud_functions/bq_table_snapshots/README.md

```diff
@@ -37,12 +37,43 @@ The following environemnt variables must be set:
 ## bq_backup_create_snapshots
 The **bq_backup_create_snapshots** cloud function will submit a BigQuery job to create a snapshot for each table in scope. This cloud function will suffix the snapshot name with the snapshot datetime to guarantee a unique name. It will also calculate and set the expiration time of the snapshot based on seconds_before_expiration. Finally, it will determine the snapshot time based on crontab_format. 
 The following environemnt variables must be set:
-* `BQ_DATA_PROJECT_ID `id of project used for BQ storage
+* `BQ_DATA_PROJECT_ID` id of project used for BQ storage
 * `BQ_JOBS_PROJECT_ID` id of project used for BQ compute
 
 
 ### About the crontab_format field
 If DATASET_1 has 500 tables, 500 Pub/Sub messages are sent, and 500 Cloud Function invocations are performed. If the Cloud Function used the current time when it creates the snapshots then these 500 snapshots will represent different points in time. To avoid this the Cloud Function will create the snapshots for the table as they were when the Cloud Scheduler job (bq-snap-start-process) was triggered. To achieve this the Cloud Function will calculate the previous interval based on **crontab_format**.
 
 
+# Deployment
+
+## Declare Variables
+### Set backend TF Bucket
+
+In [backend.tf](./terraform/backend.tf) peify a bucket for storing terraform state as an object (must be created before running terraform)
+
+### Set the following variables in the [deployment_vars.tfvars](./terraform/deployment_vars.tfvars)
+
+`project_id` project in which solution will be executed 
+
+`storage_project_id` project used for storage 
+
+`source_dataset_name` dataset for which datasets will be made 
+
+`target_dataset_name` dataset where dataset will be stored 
+
+`crontab_format` crontab for the execution of the solution
+
+`seconds_before_expiration` seconds after which snapshots will expire
+
+## Terraform Provisioning
+```
+git clone https://github.com/GoogleCloudPlatform/bigquery-utils.git
+
+cd ./bigquery-utils/tools/cloud_functions/bq_table_snapshots/terraform
+
+terraform init
+
+terraform apply --var-file=deployment_vars.tfvars --auto-approve
```

Comment:
Terraform auto loads tfvars files only when they are named terraform.tfvars. 
Recommend renaming deployment_vars.tfvars file to terraform.tfvars so that this command is cleaner:
`terraform apply --auto-approve`